packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -225,38 +226,72 @@ class _OverlayEntryWidget extends StatefulWidget {
   const _OverlayEntryWidget({
     required Key key,
     required this.entry,
+    required this.overlayState,
     this.tickerEnabled = true,
   }) : assert(key != null),
        assert(entry != null),
        assert(tickerEnabled != null),
        super(key: key);
 
   final OverlayEntry entry;
+  final OverlayState overlayState;
```

Comment:
Hm, maybe the intend would be clearer if you grab & cache it from the entry then? Or is the entry no updated yet?

(the duplications make it kind of hard to follow the code, everything seems to be passed to everything...)

Reply:
yeah I think getting the `OverlayState` from `entry` will make it easier to run into edge cases, since `entry._overlay` changes immediately. For example if `entry.remove()` is called during build, `_overlay` will be set to null but the overlay entry will stay mounted until the next frame:

https://github.com/flutter/flutter/blob/c07f7888888435fd9df505aa2efc38d3cf65681b/packages/flutter/lib/src/widgets/overlay.dart#L167-L179